source/plugin/permissions/definitions.rst

```diff
@@ -0,0 +1,181 @@
+    Role: staff
+    Owner: MyPlugin v1.2.3
+
+.. tip::
+
+    You might skip writing descriptions for some parent permission groups such as ``myPlugin.commands.teleport.worlds``
+    or ``myPlugin.commands`` as their meaning can be derived from the permission structure and the defined children
+    alone.
+
+Subject
+=======
+
+A Subject is a holder of assigned permissions it can be obtained from all CommandSources.
+Permissions can be granted or denied. Plugin authors will usually only deal with Subjects.
+
+Example
+~~~~~~~
+
+The permission plugin's config might look like this.
+
+.. code-block:: text
+    
+    users:
+      Spongie:
+        permissions:
+          - myPlugin.commands.teleport.execute
+          - myPlugin.commands.teleport.others
+          - myPlugin.commands.teleport.worlds
+          - !myPlugin.commands.teleport.worlds.protectedworld
+
+In this case the user `Spongie` was granted the permission to execute the teleport command and teleport her or other players to all worlds, , other than `protectedworld`.
+
+Inheritance
+~~~~~~~~~~~
+
+Subjects also inherit permissions from their parents, unless the given permission is defined in the subject itself.
+Following that rule if any of his parents is granted the ``myPlugin.commands.teleport`` permission the user would
+automatically inherit the ``myPlugin.commands.teleport.all`` permission allowing him to teleport all players at once.
+He still is not able to teleport himself or others to the `protectedworld` though.
+
+If neither the subject itself nor any parent subjects grant or deny a permission then it will be inherited from the
+default subject. The permission plugin/the server owner alone is allowed to define persistent defaults. 
+All other plugins are required to define `transient`/`virtual` defaults for their own permissions.
+
+.. note::
+
+    Persistent permission assignments always take precedence over transient ones.
+
+TODO: Specify persistent vs transient behaviour regarding inheritance.
+
+If neither the subject, nor any of its parent, nor the default grant or deny a permission, then it is automatically
+denied.
+
+.. note::
+
+    Order of precedence in descending order:
+    
+    * Subject himself
+    * Parent subjects
+    * Defaults
+      * Persistent (Server owner via Permission plugin)
+      * Transient (Plugins themselves)
+    * Deny permission
+
+SubjectCollections
+==================
+
+A container for subjects that can be used obtained a Subject by name.
+These are the default Subject Collections:
+
+* User/Player
+* Group 
+* System
+* Command Block
+* Role Template
+
+SubjectData
+===========
+
+TODO: The storage of the actual permissions and inheritance information.
+
+OptionSubject
+=============
+
+TODO: An inheritable String value such as a chat prefix.
+
+Contexts
+========
+
+TODO: A context (name) that may or may not apply to a `Subject` and can be used to assign permissions only in specific
```

Comment:
My best description would be:

If you consider each permission to a privilege or ability to be able to do something, a Context is the circumstances where that privilege is usable.

You might want to give a Subject permission to do something, but only when the Subject is in a certain world, or in a certain Region.

Contexts are accumulated by a Subject, and are then used by the PermissionService to decide if the Subject has a privilege or not. 

Sponge provides some contexts by default, but it is generally down to other plugins to provide additional contexts to the PermissionService, through a ContextCalculator.